[Synapse monitoring] Upgrade rollup to v3

deyaaeldeen opened the pull request:
### Packages impacted by this PR
@azure/synapse-monitoring

### Issues associated with this PR
https://github.com/Azure/azure-sdk-for-js/pull/27285

### Describe the problem that is addressed by this PR
Upgrading rollup to v3 and fixing compilation, bundling, and testing issues.

### What are the possible designs available to address the problem? If there are more than one possible design, why was the one in this PR chosen?
N/A

### Are there test cases added in this PR? _(If not, why?)_
N/A

### Provide a list of related PRs _(if any)_
N/A

### Command used to generate this PR:**_(Applicable only to SDK release request PRs)_

### Checklists
- [x] Added impacted package name to the issue description
- [ ] Does this PR needs any fixes in the SDK Generator?** _(If so, create an Issue in the [Autorest/typescript](https://github.com/Azure/autorest.typescript) repository and link it here)_
- [ ] Added a changelog (if necessary)

jeremymeng reviewed (approved):
Does it have existing recordings? Not seeing deletions in this PR.

The source looks generated by @joheredi. Not sure if we want to format them.

deyaaeldeen commented:
> Does it have existing recordings? Not seeing deletions in this PR.

Good point, deleted in https://github.com/Azure/azure-sdk-for-js/pull/27372/commits/604086502573811d3bdd3d610a7d36ee08affe79

> The source looks generated by @joheredi. Not sure if we want to format them.

I was on the fence but I guess no harm in not formatting, reverted in https://github.com/Azure/azure-sdk-for-js/pull/27372/commits/ba758d7d8c9af6be82ee0c8f6c9814f9c5dc38a2

azure-sdk commented:
**API change check**

API changes are not detected in this pull request.
